File: ios-lister/CloudStoreLister/Logic/InventoryDisplayLogic.swift
import Foundation

enum InventoryDisplayLogic {
    static func stockLabel(
        trackInventory: Bool,
        quantityOnHand: Int?,
        inStock: Bool,
        lowStock: Bool
    ) -> String {
        guard trackInventory else { return "Not tracked" }
        guard let quantityOnHand else { return "—" }
        if !inStock { return "Out of stock" }
        if lowStock { return "\(quantityOnHand) (low)" }
        return "\(quantityOnHand)"
    }

    static func priceDetail(regularPrice: Double, onSale: Bool, salePrice: Double?) -> String {
        if onSale, let salePrice {
            return String(format: "Sale $%.2f (was $%.2f)", salePrice, regularPrice)
        }
        return String(format: "$%.2f", regularPrice)
    }
}

File: android-lister/app/src/main/java/com/cloudstore/lister/domain/InventoryDisplayLogic.kt
package com.cloudstore.lister.domain

object InventoryDisplayLogic {
    fun stockLabel(
        trackInventory: Boolean,
        quantityOnHand: Int?,
        inStock: Boolean,
        lowStock: Boolean,
    ): String {
        if (!trackInventory) return "Not tracked"
        val qty = quantityOnHand ?: return "—"
        if (!inStock) return "Out of stock"
        if (lowStock) return "$qty (low)"
        return qty.toString()
    }

    fun priceDetail(regularPrice: Double, onSale: Boolean, salePrice: Double?): String {
        if (onSale && salePrice != null) {
            return "Sale $%.2f (was $%.2f)".format(salePrice, regularPrice)
        }
        return "$%.2f".format(regularPrice)
    }
}
